scripts/table_updates/update_data_table.py

```diff
@@ -423,79 +454,258 @@ def update_redact_table(
+    syn: synapseclient.Synapse,
+    master_table: pandas.DataFrame,
+    logger: logging.Logger,
+    config: dict,
+    cohort: str = "",
+) -> None:
+    """
+    This overwrites tier1a
 
     Args:
         syn (synapseclient.Synapse): synapse client connection
         master_table (pandas.DataFrame): table of all of the primary BPC tables
         logger (logging.Logger): logger object
         config (dict): config read in
+        cohort (str): cohort name
     """
-    logger.info("Custom fix in progress...")
-    # Modify the cpt_seq_date table per request
-    cpt_table_id = master_table.loc[
-        master_table["form_label"] == "Cancer Panel Test", "id"
-    ].values[0]
-    cpt_table_schema = syn.get(cpt_table_id)
-    cpt_dat_query = syn.tableQuery("SELECT cpt_genie_sample_id FROM %s" % cpt_table_id)
-    cpt_dat = download_synapse_table(
-        syn, table_id=cpt_table_id, select="cpt_genie_sample_id"
-    )
-    cpt_dat.index = cpt_dat.index.map(str)
-    cpt_dat["index"] = cpt_dat.index
-    genie_sample_dat = get_main_genie_clinical_sample_file(
+    logger.info("Overwrite tier1a variables in progress...")
+    # load GENIE BPC elements mapping table
+    column_mapping_table = utilities.download_synapse_table(syn, "syn20945902")
+    genie_patient_dat = get_main_genie_clinical_file(
         syn,
         release=config["main_genie_release_version"],
         release_files_table_synid=config["main_genie_data_release_files"],
+        form="patient_characteristics",
+        column_mapping_table=column_mapping_table,
         logger=logger,
     )
-    cpt_seq_dat = cpt_dat.merge(
-        genie_sample_dat,
-        how="left",
-        left_on="cpt_genie_sample_id",
-        right_on="SAMPLE_ID",
+    genie_sample_dat = get_main_genie_clinical_file(
+        syn,
+        release=config["main_genie_release_version"],
+        release_files_table_synid=config["main_genie_data_release_files"],
+        form="cancer_panel_test",
+        column_mapping_table=column_mapping_table,
+        logger=logger,
     )
-    cpt_seq_dat.index = cpt_seq_dat["index"]
-    cpt_seq_dat = cpt_seq_dat[["SEQ_YEAR"]]
-    cpt_seq_dat.columns = ["cpt_seq_date"]
-    cpt_seq_dat["cpt_seq_date"] = cpt_seq_dat["cpt_seq_date"].map(float_to_int)
-    syn.store(Table(cpt_table_schema, cpt_seq_dat, etag=cpt_dat_query.etag))
-    # Modify the cpt_sample_type -> map to text value
-    cpt_table_schema = syn.get(cpt_table_id)
-    cpt_dat_query = syn.tableQuery(
-        "SELECT cpt_sample_type FROM %s WHERE cpt_sample_type in (1,2,3,4,5,6,7)"
-        % cpt_table_id
+    # unlist form column in master table
+    master_table["form"] = master_table["form"].apply(lambda x: ", ".join(x))
+    # modify for patient table
+    cpt_table_id, cpt_seq_dat = update_tier1a(
+        syn,
+        "patient_characteristics",
+        master_table,
+        genie_patient_dat,
+        column_mapping_table,
+        bpc_column_list=[
```

Comment:
nit: could make this a list that gets passed in incase this list changes

Reply:
Done.